tiledb/sm/c_api/tiledb_dimension_label.cc

```diff
@@ -212,153 +212,168 @@ using tiledb::api::api_entry_with_context;
 template <auto f>
 constexpr auto api_entry = tiledb::api::api_entry_with_context<f>;
 
-capi_return_t tiledb_array_schema_add_dimension_label(
+CAPI_INTERFACE(
+    array_schema_add_dimension_label,
```

Comment:
GitHub won't let me comment on the correct line since its not part of the diff. However, all of the implementation functions in this file are in the `tiledb::common::detail` namespace which breaks the trait class generation. Swapping the namespace to `tiledb::api::` fixes the issue for me locally.